contracts/staking/compound/GoodCompoundStaking.sol

```diff
@@ -177,24 +180,30 @@ contract GoodCompoundStaking is SimpleStaking {
 		uint256 er = cToken.exchangeRateStored();
 		(uint256 decimalDifference, bool caseType) = tokenDecimalPrecision();
 		uint256 mantissa = 18 + tokenDecimal() - iTokenDecimal();
-		uint256 tokenBalance = iTokenWorthInToken(
-			iToken.balanceOf(address(this))
-		);
-		uint256 balanceInUSD = _returnTokenBalanceInUSD
-			? getTokenValueInUSD(tokenUsdOracle, tokenBalance)
-			: 0;
-		uint256 compValueInUSD = _returnTokenGainsInUSD
-			? getCompValueInUSD(
-				ERC20(nameService.getAddress("COMP")).balanceOf(address(this))
-			)
-			: 0;
+		uint256 tokenBalance =
+			iTokenWorthInToken(iToken.balanceOf(address(this)));
+		uint256 balanceInUSD =
+			_returnTokenBalanceInUSD
+				? getTokenValueInUSD(tokenUsdOracle, tokenBalance)
+				: 0;
+		uint256 compValueInUSD =
+			_returnTokenGainsInUSD
+				? getTokenValueInUSD(
+					compUsdOracle,
+					ERC20(nameService.getAddress("COMP")).balanceOf(
+						address(this)
+					)
+				)
+				: 0;
 		if (tokenBalance <= totalProductivity) {
 			return (0, 0, tokenBalance, balanceInUSD, compValueInUSD);
 		}
 		uint256 tokenGains = tokenBalance - totalProductivity;
-		uint256 tokenGainsInUSD = _returnTokenGainsInUSD
-			? getTokenValueInUSD(tokenUsdOracle, tokenGains) + compValueInUSD
-			: 0;
+		uint256 tokenGainsInUSD =
+			_returnTokenGainsInUSD
+				? getTokenValueInUSD(tokenUsdOracle, tokenGains) +
```

Comment:
are we sure about the calculations in getTokenValueInUSD and the result which is used to calculate the gas refund?

Reply:
Yes I changed getTokenValueInUSD accepts also token decimal to make sure we divide it to correct  10**decimals